WEB/Src/DependencyCollector/DependencyCollector/Implementation/AzureSdk/AzureSdkDiagnosticsEventHandler.cs

```diff
@@ -325,6 +333,49 @@ private static bool IsMessagingDependency(string dependencyType)
                          dependencyType.EndsWith(RemoteDependencyConstants.AzureServiceBus, StringComparison.Ordinal));
         }
 
+        private static void SetCosmosDbProperties(Activity activity, DependencyTelemetry telemetry)
+        {
+            string dbAccount = null;
+            string dbName = null;
+            string dbOperation = null;
+            string dbContainer = null;
+
+            foreach (var tag in activity.Tags)
+            {
+                if (tag.Key == "db.name")
+                {
+                    dbName = tag.Value;
+                }
+                else if (tag.Key == "db.operation")
+                {
+                    dbOperation = tag.Value;
+                }
+                else if (tag.Key == "net.peer.name")
+                {
+                    dbAccount = tag.Value;
+                }
+                else if (tag.Key == "db.cosmosdb.container")
+                {
+                    dbContainer = tag.Value;
+                }
+                else if (tag.Key == "db.cosmosdb.status_code")
+                {
+                    telemetry.ResultCode = tag.Value;
+                    continue;
+                }
+                else if (!tag.Key.StartsWith("db.cosmosdb.", StringComparison.Ordinal))
+                {
+                    continue;
+                }
+                
+                telemetry.Properties[tag.Key] = tag.Value;
+            }
+
+            // similar to SqlClientDiagnosticSourceListener
+            telemetry.Target = string.Join(" | ", dbAccount, dbName);
```

Comment:
ever need to account for the possibility of either of this being null?

Reply:
it should not be possible, but agree that we should be a bit more cautions here. fixed.